import "nameserver.proto";
import "status_code.proto";

package baidu.bfs;

option cc_generic_services = true;

message RemoveBlockRequest {
    optional int64 sequence_id = 1;
    optional int64 block_id = 2;
}

message RemoveBlockResponse {
    optional int64 sequence_id = 1;
    optional StatusCode status = 2;
}

service MetaServer {
    rpc AddBlock(AddBlockRequest) returns(AddBlockResponse);
    rpc SyncBlock(SyncBlockRequest) returns(SyncBlockResponse);
    rpc FinishBlock(FinishBlockRequest) returns(FinishBlockResponse);
    rpc RemoveBlock(RemoveBlockRequest) returns(RemoveBlockResponse);
}
